metals/src/main/scala/scala/meta/internal/metals/ScalafixProvider.scala

```diff
@@ -0,0 +1,147 @@
+) {
+  import ScalafixProvider._
+  private val scalafixCache = mutable.Map.empty[String, Scalafix]
+
+  def organizeImports(file: AbsolutePath): List[l.TextEdit] = {
+    val fileInput = file.toInput
+    val unsavedFile = file.toInputFromBuffers(buffers)
+    if (fileInput.text.diff(unsavedFile.text).nonEmpty) {
+      scribe.info(s"Organize imports requires saving the file first")
+      languageClient.showMessage(
+        MessageType.Warning,
+        s"Save ${file.toNIO.getFileName} before organizing imports"
+      )
+      Nil
+    } else {
+      val baseMatcher =
+        FileSystems.getDefault.getPathMatcher("glob:**.{scala,sbt}")
+      if (baseMatcher.matches(file.toNIO)) {
+        val scalafixConfPath = workspace.resolve(scalafixConfigPath)
+        val scalafixConf =
+          if (scalafixConfPath.isFile) Some(scalafixConfPath.toNIO)
+          else None
+        val scalafixEvaluation = for {
+          (scalaVersion, scalaBinaryVersion, classPath) <-
+            getScalaVersionAndClassPath(file)
+          api <- getOrUpdateScalafixCache(scalaBinaryVersion)
+          scalafixArgs = configureApi(api, scalaVersion, classPath)
+          urlClassLoaderWithExternalRule = embedded.organizeImports(
+            scalaBinaryVersion,
+            api.getClass.getClassLoader
+          )
+        } yield {
+          val scalacOption =
+            if (scalaBinaryVersion == "2.13") "-Wunused:imports"
+            else "-Ywarn-unused-import"
+
+          scalafixArgs
+            .withToolClasspath(urlClassLoaderWithExternalRule)
+            .withConfig(scalafixConf.asJava)
+            .withRules(List(organizeImportRuleName).asJava)
+            .withPaths(List(file.toNIO).asJava)
+            .withSourceroot(workspace.toNIO)
+            .withScalacOptions(Collections.singletonList(scalacOption))
+            .evaluate()
+        }
+
+        val newFileContentOpt = scalafixEvaluation
+          .flatMap(result => result.getFileEvaluations.headOption)
+          .flatMap(_.previewPatches().asScala)
+        newFileContentOpt.map(getTextEditsFrom(_, fileInput)).getOrElse(Nil)
+      } else {
+        scribe.info(
+          s"""|Could not organize import for ${file.toNIO.getFileName}. Should end with .scala or .sbt"""
+        )
+        Nil
+      }
+    }
+  }
+
+  private def getTextEditsFrom(
+      newFileContent: String,
+      input: Input
+  ): List[l.TextEdit] = {
+    val fullDocumentRange = Position.Range(input, 0, input.chars.length).toLSP
+    if (newFileContent != input.text) {
+      List(new l.TextEdit(fullDocumentRange, newFileContent))
+    } else {
+      Nil
+    }
+  }
+
+  private def getScalaVersionAndClassPath(
+      file: AbsolutePath
+  ): Option[(ScalaVersion, ScalaBinaryVersion, util.List[Path])] =
+    for {
+      identifier <- buildTargets.inverseSources(file)
+      scalacOptions <- buildTargets.scalacOptions(identifier)
+      scalaBuildTarget <- buildTargets.scalaInfo(identifier)
```

Comment:
```suggestion
      scalaBuildTarget <- buildTargets.scalaTarget(identifier)
```
this should give you a nicer api overall.